### ISSUE TITLE
Arity specifications for discouraged-var produce false negatives

### ISSUE THREAD

**Issue by imrekoszo** [clj-kondo/clj-kondo#2465]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

```
; clj-kondo -version | head -n 1
clj-kondo v2025.01.16
```

**platform**

macos, homebrew `borkdude/brew/clj-kondo`

**problem**

Some warnings are missing, which should be triggered by specifying arities for discouraged-var.

Varargs seem to work well on the other hand, and so do arities that were not specified:

```
; echo '(map vector [2] [3] [4] [5])' | clj-kondo --config '{:linters {:discouraged-var {clojure.core/map {:arities #{:varargs}}}}}' --lint -
<stdin>:1:1: warning: Discouraged var: clojure.core/map
linting took 61ms, errors: 0, warnings: 1

; echo '(map vector)' | clj-kondo --config '{:linters {:discouraged-var {clojure.core/map {:arities #{2 3 4 :varargs}}}}}' --lint -
linting took 57ms, errors: 0, warnings: 0
```

**repro**

```
; echo '(map vector [2]) (map vector [2] [3]) (map vector [2] [3] [4])' | clj-kondo --config '{:linters {:discouraged-var {clojure.core/map {:arities #{2 3 4}}}}}' --lint -
linting took 68ms, errors: 0, warnings: 0
```

**expected behavior**

3 warnings should be produced
